File: apina-core/src/main/java/fi/evident/apina/model/settings/ImportDefinition.java
package fi.evident.apina.model.settings;

import fi.evident.apina.model.type.ApiTypeName;

import java.util.Set;
import java.util.TreeSet;

import static java.util.Collections.unmodifiableSet;
import static java.util.Objects.requireNonNull;

public final class ImportDefinition {

    private final String moduleName;
    private final Set<ApiTypeName> types = new TreeSet<>();

    public ImportDefinition(String moduleName) {
        this.moduleName = requireNonNull(moduleName);
    }

    public String getModuleName() {
        return moduleName;
    }

    public Set<ApiTypeName> getTypes() {
        return unmodifiableSet(types);
    }

    public void addType(ApiTypeName type) {
        types.add(requireNonNull(type));
    }
}

File: apina-core/src/main/kotlin/fi/evident/apina/model/settings/ImportDefinition.kt
package fi.evident.apina.model.settings

import fi.evident.apina.model.type.ApiTypeName
import java.util.*

class ImportDefinition(val moduleName: String) {

    private val _types = TreeSet<ApiTypeName>()

    val types: Set<ApiTypeName>
        get() = _types

    fun addType(type: ApiTypeName) {
        _types.add(type)
    }
}
